core/src/main/java/org/kiwix/kiwixmobile/core/search/DefaultAdapter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.search;
+
+import android.content.Context;
+import android.text.Html;
+import android.text.style.ClickableSpan;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+public class DefaultAdapter extends RecyclerView.Adapter<DefaultAdapter.ViewHolder> {
+
+  private Context context;
+  List<String> searchList;
+
+  interface CLickListener {
```

Comment:
An Interface definition should not be inbetween fields, at the top of the class is probably for the best